lib/lookup/src/lookup_v2/mod.rs

```diff
@@ -22,12 +22,45 @@ impl<'de> Deserialize<'de> for OwnedPath {
 }
 
 impl Serialize for OwnedPath {
-    fn serialize<S>(&self, _serializer: S) -> Result<S::Ok, S::Error>
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
         S: Serializer,
     {
-        // TODO: Implement canonical way to serialize segments.
-        todo!()
+        if self.segments.is_empty() {
+            serializer.serialize_str(".")
+        } else {
+            let path = self
+                .segments
+                .iter()
+                .map(|segment| match segment {
+                    OwnedSegment::Field(field) => {
+                        let needs_quotes = field
+                            .chars()
+                            .any(|c| !matches!(c, 'A'..='Z' | 'a'..='z' | '_' | '0'..='9' | '@'));
+                        if needs_quotes {
+                            let mut string = String::from(r#".""#);
+                            string.reserve(field.as_bytes().len() + 1);
```

Comment:
Not a big deal since this isn't likely to be on a hot path, but it seems like you could have less allocations by doing `String::with_capacity` and then push on the `."`?

Reply:
Yeah that seems reasonable. The `push` is likely way cheaper than the first (albeit small) allocation

Reply:
Applied your suggestion in 5a06ba592bd783dd26baabb1f083e2973a828770